Pressing Enter to close Start menu in Windows 10 no longer results in NVDA announcing search field text. re #7370.

josephsl opened the pull request:
Resolves the following:

* Fixes #7370: In Windows 10's start menu, pressing Enter to close the menu after entering search term results in NVDA announcing search field text.
Suggested what's new entry:

section: bug fixes


In start menu in Windows 10, pressing Enter to close the start menu after a search no longer causes NVDA to announce search text.

Thanks.

josephsl committed:
UIA/Search field: do not announce search field entry when searchui (start menu) is closing. re #7370.

due to use of announceNewlineText, when start menu closes and when Enter is pressed, search field value will be announced, which sometimes leads to side effects such as core freeze. Address this through initOverClass in UIA.SearchField that turns this flag off if and only if we're in start menu in Windows 10.

derekriemer commented on source/NVDAObjects/UIA/__init__.py:
```diff
@@ -1468,6 +1468,10 @@ class SearchField(EditableTextWithSuggestions, UIA):
 
+	def initOverlayClass(self):
+		# #7370: do not announce text when start menu (searchui) closes.
+		self.announceNewLineText = self.appModule.appName != "searchui"
```
Would creating a new overlay just for this be better?

derekriemer reviewed (commented):
Questions

michaelDCurran reviewed (changes requested):
I agree with Derek. Either make this into its own overlay class in searchUI, or perhaps even just place this variable on the SearchField class no matter if it is searchUI or not. For now at least, this class is only used for single-line edit fields where  speaking on enter makes no sense. Obviously I do acknowledge that in the future this may change.

josephsl commented:
Hi, when I set announceNewlineText inside SearchField class, it returns True even though I said False, yet the app module method works. Modification on the way.

 

From: Michael Curran [mailto:notifications@github.com] 
Sent: Monday, July 10, 2017 11:16 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Pressing Enter to close Start menu in Windows 10 no longer results in NVDA announcing search field text. re #7370. (#7375)

 

@michaelDCurran requested changes on this pull request.

I agree with Derek. Either make this into its own overlay class in searchUI, or perhaps even just place this variable on the SearchField class no matter if it is searchUI or not. For now at least, this class is only used for single-line edit fields where speaking on enter makes no sense. Obviously I do acknowledge that in the future this may change.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/7375#pullrequestreview-49101882> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkGp82D7_yBa5NijKxufZ5tppoiuKks5sMxM0gaJpZM4OTrW3> .

josephsl committed:
SearchField/searchui: an overlay class just for Start menu version that disables announcement of newline text. re #7370.

Reviewed by Mick Curran (NV Access): better to create an overlay class for this just for searchui. This also allows easy modifications in the future if the need arises.
